src/js/thirdparty/node-fetch.ts

```diff
@@ -0,0 +1,96 @@
+import type * as s from "stream";
+
+const { Headers, Request, Response: WebResponse, Blob, File = Blob, FormData } = globalThis as any;
+const nativeFetch = Bun.fetch;
+
+const { Readable } = require("node:stream");
+
+class Response extends WebResponse {
+  _body: any;
+
+  get body() {
+    return this._body ?? (this._body = Readable.fromWeb(super.body));
+  }
+}
+
+/**
+ * `node-fetch` works like the browser-fetch API, except it's a little more strict on some features,
+ * and uses node streams instead of web streams.
+ *
+ * It's overall a positive on speed to override the implementation, since most people will use something
+ * like `.json()` or `.text()`, which is faster in Bun's native fetch, vs `node-fetch` going
+ * through `node:http`, a node stream, then processing the data.
+ */
+async function fetch(url: any, init?: RequestInit & { body?: any }) {
+  // input node stream -> web stream
+  let body: s.Readable | undefined = init?.body;
+  if (body) {
+    const chunks: any = [];
+    if (body instanceof Readable) {
+      for await (const chunk of body) {
+        chunks.push(chunk);
+      }
+      init = { ...init, body: new Blob(chunks) };
+    }
+  }
+
+  const response = await nativeFetch(url, init);
+  Object.setPrototypeOf(response, Response.prototype);
```

Comment:
how slow is this prototype switch? should this be a subclass that implements all the getters or something else?

A good thing about this is it makes the wrapper very minimal in terms of # of js code.

Reply:
pretty bad but not as bad as the network call